data_science/palette/notebooks/00 - RGB, LAB, and human colour perception.ipynb

```diff
@@ -0,0 +1,133 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# RGB and CIE\n",
+    "In this project we'll regularly need to calculate distances between pixels in colour space as a proxy for the visual difference between the colours. The simplest way of doing this is to calculate the cosine/euclidian/l2 distance, as written below in maths and python:\n",
+    "\n",
+    "${\\displaystyle \\Delta C ={\\sqrt {(R_{2}-R_{1})^{2}+(G_{2}-G_{1})^{2}+(B_{2}-B_{1})^{2}}}}$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def colour_distance(colour_1, colour_2):\n",
+    "    return sum([(a - b) ** 2 for a, b in zip(colour_1, colour_2)]) ** 0.5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The red, green, and blue channels available to us in RGB space are ideally suited for representing colour on pixelated screens. However, our goal is to represent the _differences_ between colours, and RGB isn't ideal for this. It's now [pretty well established](https://en.wikipedia.org/wiki/Color_difference) that euclidian distances in RGB space are a bad representation of the distances that our eyes see.\n",
+    "By stretching the RGB dimensions by different amounts, we can better approximate that difference:\n",
+    "\n",
+    "${\\displaystyle {\\sqrt {2\\times \\Delta R^{2}+4\\times \\Delta G^{2}+3\\times \\Delta B^{2}}}}$\n",
+    "\n",
+    "Again, here's the python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def colour_distance(colour_1, colour_2):\n",
+    "    r_1, g_1, b_1 = colour_1\n",
+    "    r_2, g_2, b_2 = colour_2\n",
+    "\n",
+    "    return (2 * (r_1 - r_2) ** 2 +\n",
+    "            4 * (g_1 - g_2) ** 2 +\n",
+    "            3 * (b_1 - b_2) ** 2) ** 0.5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can improve further by adding some extra weirdness to the red and blue channels at the end\n",
```

Comment:
“extra weirdness” is not a good term to use in docs :-P

How about “a better approximation is”